src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,177 @@
+from math import ceil
+from typing import Any, List, Mapping, Optional, Sequence, Union
+
+import tiledb
+from tiledb.cloud import dag
+from tiledb.cloud.dag.mode import Mode
+from tiledb.cloud.utilities import as_batch
+from tiledb.cloud.utilities import get_logger_wrapper
+from tiledb.cloud.vcf.ingestion import find_uris_udf
+
+DEFAULT_RESOURCES = {"cpu": "2", "memory": "8Gi"}
+DEFAULT_FILE_INGESTION_NAME = "file-ingestion"
+DEFAULT_BATCH_SIZE = 100
+
+
+def chunk_results_udf(
+    udf_results: Sequence[Sequence[str]], batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+) -> List[List[str]]:
+    """
+    Flatten and break a list of udf results into batches of a specified size.
+
+    :param udf_results: An iterable of iterables containing UDF results.
+    :param batch_size: Resulting chunk size, defaults to 100.
+    :return List[List[str]]: A list of chunks as lists.
+    """
+    # Reduce batch size if there are fewer sample URIs
+    batch_size = min(batch_size, len(udf_results))
+    num_chunks = ceil(len(udf_results) / batch_size)
+
+    flattened_results = [result for udf_result in udf_results for result in udf_result]
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
+        for n in range(num_chunks)
+    ]
+
+
+def ingest_files_udf(
+    dataset_uri: str,
+    file_uris: Sequence[str],
+    *,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Ingest files.
+
+    :param dataset_uri: The dataset URI.
+    :param file_uris: An iterable of file URIs.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: Verbose logging, defaults to False.
+    """
+    logger = get_logger_wrapper(verbose)
+
+    for file_uri in file_uris:
+        filename = file_uri.replace(",", "").replace(" ", "_")
+        filestore_array_uri = f"{dataset_uri}/{filename}"
+        filestore_array_s3_uri = f"s3{filestore_array_uri.split('s3')[1]}"
+        namespace = (
+            namespace or filestore_array_uri.split("tiledb://")[1].split("/s3")[0]
```

Comment:
Why do we assume that all `tiledb://` are of the format `tiledb://namespace/s3://..`?